core/controllers/base_test.py

```diff
@@ -269,6 +270,21 @@ def test_i18n_keys(self):
                     self.log_line('- %s' % key)
                 self.log_line('')
 
+    def test_alphabetic_i18n_keys(self):
+        """Tests that the keys of all i18n json files are arranged in
+        alphabetical order."""
+        filenames = os.listdir(
+            os.path.join(os.getcwd(), self.get_static_asset_filepath(),
+                         'assets', 'i18n'))
+        for filename in filenames:
+            with open(os.path.join(os.getcwd(), 'assets', 'i18n', filename),
+                      mode='r') as f:
+                next_line = next(f)
+                for line in f:
+                    line = line.replace("{", "")
```

Comment:
It's weird that you're doing this for every line, when it should only affect the first and the last lines. Could you follow the following algorithm instead:

- Use f.readlines() to get a list of all the lines.
- Assert that the first element in this list is a single '{', and remove it.
- Assert that the last element in this list is a single '}', and remove it.
- Create a new list with one element for each of the remaining lines. That element should contain just the key that's enclosed in quotation marks. For example, if the original line is `"I18N_SIDEBAR_FORUM": "Foro",` then the corresponding entry in the new list should be `I18N_SIDEBAR_FORUM`. This also addresses potential issues involving comparing unicode characters, since the characters comprising the keys are always guaranteed to be ascii.
- Finally, check that the new list is in sorted order.

Thanks!